File: javascript/src/infraestrutura/ConsultaRepositoryMemoria.js
import { ConsultaRepository } from '../porta/ConsultaRepository.js';

/**
 * LSP: substitui a versao Postgres sem quebrar nada. E por isso que da pra rodar
 * o sistema inteiro em teste, sem banco.
 */
export class ConsultaRepositoryMemoria extends ConsultaRepository {
  #consultas = [];

  salvar(consulta) {
    if (!this.#consultas.includes(consulta)) {
      this.#consultas.push(consulta);
    }
  }

  buscarPorId(id) {
    const achada = this.#consultas.find((c) => c.id === id);
    if (!achada) {
      throw new Error(`Consulta nao encontrada: ${id}`);
    }
    return achada;
  }

  buscarPorMedicoNoDia(medico, dia) {
    return this.#consultas.filter(
      (c) => c.medico === medico && c.dataHora.toDateString() === dia.toDateString(),
    );
  }
}

File: java/src/br/unifor/facilitador/infraestrutura/ConsultaRepositoryMemoria.java
package br.unifor.facilitador.infraestrutura;

import br.unifor.facilitador.dominio.Consulta;
import br.unifor.facilitador.dominio.Medico;
import br.unifor.facilitador.porta.ConsultaRepository;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

/**
 * LSP: substitui a versao Postgres sem quebrar nada. E por isso que da pra rodar
 * o sistema inteiro em teste, sem banco.
 */
public class ConsultaRepositoryMemoria implements ConsultaRepository {

    private final List<Consulta> consultas = new ArrayList<>();

    @Override
    public void salvar(Consulta c) {
        if (!consultas.contains(c)) {
            consultas.add(c);
        }
    }

    @Override
    public Consulta buscarPorId(UUID id) {
        return consultas.stream()
                .filter(c -> c.getId().equals(id))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("Consulta nao encontrada: " + id));
    }

    @Override
    public List<Consulta> buscarPorMedicoNoDia(Medico m, LocalDate dia) {
        List<Consulta> achadas = new ArrayList<>();
        for (Consulta c : consultas) {
            if (c.getMedico() == m && c.getDataHora().toLocalDate().equals(dia)) {
                achadas.add(c);
            }
        }
        return achadas;
    }
}
